app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java

```diff
@@ -343,4 +352,55 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
         }
     }
 
+    /**
+     * Perform secure key generation in an AsyncTask.
+     */
+    private void onKeyGenerationSlideShown() {
+        mNextButton.setVisibility(View.GONE);
+        KeyGenerationTask keyGenerationTask = new KeyGenerationTask(this);
+        keyGenerationTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    /**
+     * Sets up the initial configuration and generates secure keys.
+     */
+    private static class KeyGenerationTask extends AsyncTask<Void, Void, Void> {
+        private WeakReference<FirstStartActivity> refFirstStartActivity;
```

Comment:
Why does this needs to be a week ref? Why can't it just be a normal reference?

Reply:
I've explained this in an earlier PR, I think it was one of those "static field leaks PRs". It's Android best coding practice, nothing more. This prevents lint warnings.